File: questions/OBI/Distinto/distinto.py
# Feito por Kaio Henrique

n = int(input())

I = []
for i in range(n):
    I.append(int(input()))


interval = set()
max_interval = 0, start = 0

for end in range(len(I)):
    while I[end] in interval:
        interval.remove(I[start])
        start += 1
    interval.add(I[end])
    size = end - start + 1
    max_interval = max(max_interval, size)
    
print(max(max_interval, size))

# Código testado e aprovado com corretor automático usando: https://github.com/brasilicioh/Solucoes-OBI/
# Max runtime: 0.27s

File: questions/OBI/Distinto/distinto.rs
// Feito por Guilherme Aleixo

use std::collections::HashSet;

fn main() {
    let stdin = std::io::stdin();

    let mut n = String::new();
    stdin.read_line(&mut n).unwrap();

    let n: usize = n.trim().parse().unwrap();

    let mut i = vec![0; n];
    let mut interval = HashSet::new();


    for index in 0..i.len() {
        let mut number = String::new();

        stdin.read_line(&mut number).unwrap();
        let number = number.trim().parse().unwrap();

        i[index] = number;
    };

    let mut max_interval = 0;
    let mut start = 0;

    for end in 0..i.len() {
        while interval.contains(&i[end]) {
            interval.remove(&i[start]);
            start += 1;
        }

        interval.insert(&i[end]);
        max_interval = std::cmp::max(max_interval, end - start + 1);
    }

    println!("{max_interval}");
}

// Código testado e aprovado com corretor automático usando: https://github.com/brasilicioh/Solucoes-OBI/
// Max runtime: 0.039s;